### PR TITLE
step 7

### PR DESCRIPTION
**Pull request by andriimikhaylovskiy**

### CODE REVIEW

File: src/main/java/mate/academy/spring/boot/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+@Service
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepo;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        if (userRepo.existsByEmail(requestDto.email())) {
+            throw new RegistrationException("This user can't be registered");
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/spring/boot/service/impl/UserServiceImpl.java]: Let's make message more specify: 
```suggestion
            throw new RegistrationException(String.format("User with this email: %s already exists", request.email());
```

File: src/main/java/mate/academy/spring/boot/validation/FieldsValueMatchValidator.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    private String field;
+    private String fieldMatch;
+
+    public void initialize(FieldsValueMatch constraintAnnotation) {
+        this.field = constraintAnnotation.field();
+        this.fieldMatch = constraintAnnotation.fieldMatch();
+    }
+
+    public boolean isValid(Object value, ConstraintValidatorContext context) {
+
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/spring/boot/validation/FieldsValueMatchValidator.java]:
```suggestion
```

Comment (Elena-Bruyako) [reply src/main/java/mate/academy/spring/boot/validation/FieldsValueMatchValidator.java]: Remove all redundant lines

File: src/main/java/mate/academy/spring/boot/validation/FieldsValueMatchValidator.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        Object fieldValue = new BeanWrapperImpl(value).getPropertyValue(field);
+        Object fieldMatchValue = new BeanWrapperImpl(value).getPropertyValue(fieldMatch);
+
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/spring/boot/validation/FieldsValueMatchValidator.java]:
```suggestion
      return Objects.equals(field, fieldMatch);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's fix some comments.

**Review by MAMentorFX — APPROVED** [review]

**Pull request merged**